File: my-folder/0001-two-sum/solution.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        HashMap<Integer,Integer> numberCountHashMap=new HashMap<Integer,Integer>();
        int[] indices=new int[2];
        for(int i=0;i<nums.length;i++)
        {
            int resTarget=target-nums[i];
            if(numberCountHashMap.containsKey(resTarget))
            {
              
                indices[0]=numberCountHashMap.get(resTarget);
                indices[1]=i;  
                break;
            }
            else
            {
            
               numberCountHashMap.put(nums[i],i); 
            }
        }
        return indices;
    }
}

File: my-folder/0001-two-sum/solution.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        var indexMap=mutableMapOf<Int,Int>()
        nums.forEachIndexed{ i,v ->
        val remainder=target-v
        if(remainder in indexMap)
         return intArrayOf(i,indexMap[remainder]!!)
        else
          indexMap[v]=i
        }
        return intArrayOf()
    }
}
